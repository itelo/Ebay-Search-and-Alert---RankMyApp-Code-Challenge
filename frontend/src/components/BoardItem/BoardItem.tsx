import React from "react";
import { Theme, createStyles, makeStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    image: {
      width: 96,
      height: 96,
      objectFit: "cover",
      borderRadius: 4,
      margin: theme.spacing(0, 1, 0, 0)
    }
  })
);

type BoardItemProps = {
  title: string;
  price: string;
  imageURL: string;
};

const BoardItem: React.FC<BoardItemProps> = props => {
  const classes = useStyles();

  return (
    <Grid
      container
      direction="row"
      onClick={() => {
        alert("pressed item");
      }}
    >
      <Grid item>
        <img src={props.imageURL} className={classes.image} />
      </Grid>
      <Grid item xs>
        <Grid
          container
          direction="column"
          justify="space-between"
          alignContent="flex-end"
        >
          <Grid item xs>
            <Typography variant="body1" component="p">
              {props.title}
            </Typography>
            <Typography variant="h6" component="p">
              R$ {props.price}
            </Typography>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  );
};

export default BoardItem;
