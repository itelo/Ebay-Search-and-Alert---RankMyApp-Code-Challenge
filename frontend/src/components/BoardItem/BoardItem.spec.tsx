import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import "@testing-library/dom";
import BoardItem from "./BoardItem";

afterEach(cleanup);

describe("<BoardItem />", () => {
  it("test basic rendering without image", () => {
    const mockFn = jest.fn();

    const { getByTestId, queryByTestId, asFragment } = render(
      <BoardItem title="board-item-title" price="12.00" onClick={mockFn} />
    );

    expect(getByTestId("title-tag")).toHaveTextContent("board-item-title");
    expect(getByTestId("price-tag")).toHaveTextContent("US $12.00");
    expect(queryByTestId("image-tag")).toBeNull();

    expect(asFragment()).toMatchSnapshot();
  });

  it("test basic rendering with image", () => {
    const mockFn = jest.fn();

    const { getByTestId, asFragment } = render(
      <BoardItem
        imageURL="http://www.someimage.com"
        title="board-item-title"
        price="12.00"
        onClick={mockFn}
      />
    );

    expect(getByTestId("title-tag")).toHaveTextContent("board-item-title");
    expect(getByTestId("price-tag")).toHaveTextContent("US $12.00");
    expect(getByTestId("price-tag")).toHaveTextContent("US $12.00");
    expect(getByTestId("image-tag")).toHaveAttribute(
      "src",
      "http://www.someimage.com"
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("test fire event click", () => {
    const mockFn = jest.fn();

    const { getByTestId } = render(
      <BoardItem
        imageURL="http://www.someimage.com"
        title="board-item-title"
        price="12.00"
        onClick={mockFn}
      />
    );

    fireEvent.click(getByTestId("button-tag"));

    expect(mockFn).toHaveBeenCalled();
  });
});
