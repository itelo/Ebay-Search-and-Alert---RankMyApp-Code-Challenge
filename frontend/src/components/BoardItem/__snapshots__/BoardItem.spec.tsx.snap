// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoardItem /> test basic rendering with image 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container"
    data-testid="button-tag"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <img
        alt="IMAGE_OF_PRODUCT"
        class="makeStyles-image-135"
        data-testid="image-tag"
        src="http://www.someimage.com"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-content-xs-flex-end MuiGrid-justify-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
            data-testid="title-tag"
          >
            board-item-title
          </p>
          <p
            class="MuiTypography-root MuiTypography-h6"
            data-testid="price-tag"
          >
            US $12.00
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<BoardItem /> test basic rendering without image 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container"
    data-testid="button-tag"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-content-xs-flex-end MuiGrid-justify-xs-space-between"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
            data-testid="title-tag"
          >
            board-item-title
          </p>
          <p
            class="MuiTypography-root MuiTypography-h6"
            data-testid="price-tag"
          >
            US $12.00
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
